### ISSUE TITLE
[a11y]: TileGroup impossible to add aria-label to fieldset

### ISSUE THREAD

**Issue by trzaskakarol** [carbon-design-system/carbon#19684]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.84.0

### React version

19.1.0

### Automated testing tool and ruleset

RTL Tests accessibility

### Description

Unable to add an aria-label attribute to a <fieldset> element within the TileGroup component.

### Reproduction/example

https://stackblitz.com/edit/github-trzxrikd?file=src%2FApp.jsx

### Steps to reproduce

1. Create any component with TileGroup component within it
2. Try to add aria-label
3. Check whether it's added in developer tools

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
